libsigner/src/events.rs

```diff
@@ -238,15 +238,14 @@ impl EventStopSignaler for SignerStopSignaler {
                 body.len(),
                 body
             );
-            match stream.write_all(req.as_bytes()) {
-                Err(e) => error!("Failed to send shutdown request: {}", e),
-                _ => (),
-            };
+            if let Err(e) = stream.write_all(req.as_bytes()) {
+                error!("Failed to send shutdown request: {}", e);
+            }
         }
     }
 }
 
-impl EventReceiver for SignerEventReceiver {
+impl<T: StacksMessageCodec + Clone> EventReceiver<T> for SignerEventReceiver<T> {
```

Comment:
Would it be cleaner to just have a defined trait that is `StacksMessageCodec + Clone`? Especially because if we need any more fields or implementations for it, we'll probably end up doing that anyways.

Reply:
Hm I will give it a try to see what happens :)

Reply:
I created a SignerEventTrait which enforces StacksMessageCodec + Send + Debug + Clone though it does restrict a few things more than they probably need to be. Let me know what you think.